pkg/apis/monitoring/v1/types.go

```diff
@@ -86,15 +86,15 @@ type PrometheusList struct {
 type PrometheusSpec struct {
 	// PodMetadata configures Labels and Annotations which are propagated to the prometheus pods.
 	PodMetadata *EmbeddedObjectMetadata `json:"podMetadata,omitempty"`
-	// ServiceMonitors to be selected for target discovery. *Deprecated:* if
+	// ServiceMonitors to be selected for target discovery. *Warning:* if
```

Comment:
Hmm, I'm not sure this is exactly what we want to get across. The behavior of leaving both values unset allowing the entire config to be managed is what is indeed deprecated. It's just that not the entire field is deprecated, just that one behavior.

Reply:
should that be done in the validating webhook instead, then?

Reply:
Well, it's only deprecated, it still works as we speak. When we remove the behavior entirely we'll remove this note.

Reply:
How about `s/Warning/NOTE`? This should suffice the use case we're aiming to target, i.e., alerting the reader in some manner while avoiding false positives by the different tools that parse this information.